src/main/java/org/jitsi/videobridge/OctoChannel.java

```diff
@@ -0,0 +1,317 @@
+
+import java.net.*;
+import java.util.*;
+
+/**
+ * Represents an Octo channel, used for bridge-to-bridge communication.
+ *
+ * @author Boris Grozev
+ */
+public class OctoChannel
+    extends RtpChannel
+{
+    /**
+     * The {@link Logger} used by the {@link RtpChannel} class to print debug
+     * information. Note that instances should use {@link #logger} instead.
+     */
+    private static final Logger classLogger
+        = Logger.getLogger(OctoChannel.class);
+
+    /**
+     * The Octo ID of the conference, configured in {@link Conference} as the
+     * global ID ({@link Conference#getGid()}).
+     */
+    private final String conferenceId;
+
+    /**
+     * The {@link MediaType} for this channel.
+     */
+    private final MediaType mediaType;
+
+    /**
+     * The {@link TransportManager} if this channel.
+     */
+    private OctoTransportManager transportManager;
+
+    /**
+     * The {@link org.ice4j.socket.DatagramPacketFilter} which (only) accepts
+     * RTP packets for this channel.
+     */
+    private RtpChannelDatagramFilter rtpFilter
+        = new OctoDatagramPacketFilter(false);
+
+    /**
+     * The {@link org.ice4j.socket.DatagramPacketFilter} which (only) accepts
+     * RTCP packets for this channel.
+     */
+    private RtpChannelDatagramFilter rtcpFilter
+        = new OctoDatagramPacketFilter(true);
+
+    /**
+     * The {@link Logger} to be used by this instance to print debug
+     * information.
+     */
+    private final Logger logger;
+
+    /**
+     * Initializes a new <tt>Channel</tt> instance which is to have a specific
+     * ID. The initialization is to be considered requested by a specific
+     * <tt>Content</tt>.
+     *
+     * @param content the <tt>Content</tt> which is initializing the new
+     * instance
+     * @param id the ID of the new instance. It is expected to be unique
+     * within the
+     * list of <tt>Channel</tt>s listed in <tt>content</tt> while the new
+     * instance
+     * is listed there as well.
+     * @throws Exception if an error occurs while initializing the new instance
+     */
+    public OctoChannel(Content content, String id)
+        throws Exception
+    {
+        super(
+                content, id, null /*channelBundleId*/,
+                OctoTransportManager.NAMESPACE, false /*initiator*/);
+
+        conferenceId = content.getConference().getGid();
+        mediaType = content.getMediaType();
+        logger
+            = Logger.getLogger(classLogger, content.getConference().getLogger());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void describe(ColibriConferenceIQ.ChannelCommon commonIq)
+    {
+        super.describe(commonIq);
+
+        // TODO: uncomment after the mvn versions are updated
+        //commonIq.setType(ColibriConferenceIQ.OctoChannel.TYPE);
+    }
+
+    /**
+     * @return The Octo ID of the conference.
+     */
+    public String getConferenceId()
+    {
+        return conferenceId;
```

Comment:
is this not implemented in any of the channel base classes?  if not, why does octo channel need it and not the others?

Reply:
Oh, nice catch. This is Octo-specific, and is not the same thing as the regular ID of the conference. It wasn't meant to be public. I'll re-work it somehow.